### PR TITLE
Remove settings spread in client side registration.

### LINKED ISSUE #166: Issues with settings spread and translations
All of the examples of registering blocks use the following approach to registering blocks.

```js
import json from '../block.json';
import edit from './edit';
import save from './save';

const { name, ...settings } = json;

registerBlockType( name, {
	...settings,
	edit,
	save,
} );

```
It was pointed out in this [comment](https://github.com/WordPress/gutenberg-examples/pull/163#discussion_r732688814) that spreading the `settings` object is not technically required as this is done on the server-side for blocks registered. It should be used to override those server-side values or in the case of registering a JavaScript-only block.

Based on a [conversation ](https://wordpress.slack.com/archives/C02QB2JS7/p1634741271051600)with @gziolo it also seems that if the settings are passed as they are defined in `block.json` ( as in the approach above ) any translations will not be processed and would need to be provided in that object.

### PR DESCRIPTION
**Pull request by ryanwelcher**
This PR removes the approach outline and closes #166

### CODE REVIEW

**Review by mkaz — APPROVED** [review]
Looks good, a bit simpler and hopefully less confusing if people are not familiar with the spread operator yet. 👍

**Review by gziolo — APPROVED** [review]

File: 01-basic-esnext/src/index.js [hunk 1]
```diff
@@ -17,13 +17,12 @@ export const blockStyle = {
 	padding: '20px',
 };
 
-// Destructure the json file to get the name and settings for the block
+// Destructure the json file to get the name of the block
 // For more information on how this works, see: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment
-const { name, ...settings } = json;
+const { name } = json;
```
Comment (gziolo) [inline 01-basic-esnext/src/index.js:22]: Would it work if we destructure while importing?

```js
import { name } from '../block.json';
```

We no longer need to use rest operator for settings, so it should be enough.

Comment (ryanwelcher) [reply 01-basic-esnext/src/index.js:22]: It does seem to work in the sense of everything compiles and the block works.  The only issue is that both the `start` and `build` command output the following warning:

`Should not import the named export 'name' (imported as 'name') from default-exporting module (only default export is available soon)`

I feel like we should leave it as-is for two reasons:

1. These are the official examples and having the first one trigger a warning could be confusing for users. The expectation would be that these just work and could lead to the user feeling like they did something wrong.
2.  Being verbose about importing and then destructuring out the `name` variable is much clearer for those that are new to destructuring. I think these examples should lean towards teaching which may come at the cost of more lines of code.

Comment (gziolo) [reply 01-basic-esnext/src/index.js:22]: I wasn't sure about that. There is an official proposal for JSON imports in JavaScript that should fix that eventually:

https://2ality.com/2021/06/json-modules.html

Last time I checked, it was at stage 3.

**Pull request merged**